CMakeLists.txt

```diff
@@ -1,67 +1,44 @@
+#
+# Project configuration
+#
 cmake_minimum_required(VERSION 3.13)  # todo: what is the minimum version required?
-
 project(libdisplaydevice
         DESCRIPTION "Library to modify display devices."
-        HOMEPAGE_URL "https://app.lizardbyte.dev")
+        HOMEPAGE_URL "https://app.lizardbyte.dev"
+        LANGUAGES CXX)
 
 set(PROJECT_LICENSE "GPL-3.0")
+set(CMAKE_CXX_STANDARD 20)
 
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     message(STATUS "Setting build type to 'Release' as none was specified.")
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
 endif()
 
-set(CMAKE_CXX_STANDARD 20)
-
-# set the module path, used for includes
-set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-
-# options
+#
+# Project optional configuration
+#
 option(BUILD_TESTS "Build tests" ON)
 option(CODE_QL_ANALYSIS "Build is for codeql analysis" OFF)
 
-# auto-set CODE_QL_ANALYSIS based on env variable
-if(DEFINED ENV{GITHUB_CODEQL_BUILD})
-    set(CODE_QL_ANALYSIS ON)
-endif()
-
-# include directories (for including header files)
-include_directories("${CMAKE_SOURCE_DIR}")
-
-# glob src files, excluding platf directory
-file(GLOB_RECURSE DD_COMMON_TARGET_FILES "${CMAKE_SOURCE_DIR}/src/*.cpp")
-list(FILTER DD_COMMON_TARGET_FILES EXCLUDE REGEX ".*/platf/.*")
-
-if(WIN32)
-    file(GLOB_RECURSE DD_PLATFORM_TARGET_FILES "${CMAKE_SOURCE_DIR}/src/platf/windows/*.cpp")
-    include(${CMAKE_MODULE_PATH}/windows.cmake)
-elseif(APPLE)
-    if(NOT CODE_QL_ANALYSIS)
-        message(FATAL_ERROR "MacOS is not supported yet.")
+#
+# Only do these if this is the main project, and not if it is included through add_subdirectory
+#
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```

Comment:
Apparently this is the latest trend in Modern CMake. I have no strong opinion about this this.

As for removing fatals, I don't think they are needed? We will have a `noop` behavior for unsupported platforms instead of complicating the CMakeLists logic (at least that's what I have envisioned initially).

Reply:
Works for me. You can remove the whole use of the CODE_QL_ANALYSIS option.